BarrysPeanutsAkkaJava/src/main/java/barryspeanuts/App.java

```diff
@@ -0,0 +1,104 @@
+package barryspeanuts;
+
+import akka.actor.ActorRef;
+import akka.actor.ActorSystem;
+import akka.actor.Props;
+import barryspeanuts.actor.ShoppingCartActor;
+import barryspeanuts.model.Address;
+import barryspeanuts.model.CreditCard;
+import barryspeanuts.model.Customer;
+import barryspeanuts.model.PurchaseItem;
+import java.math.BigDecimal;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.UUID;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class App {
+  public static void main(String[] args) {
+    Logger logger = LoggerFactory.getLogger(ShoppingCartActor.class);
+
+    logger.info("Starting Barry's Gourmet Peanuts.");
+
+    akka.actor.ActorSystem system = ActorSystem.create("BarrysPeanutsSystem");
+
+    Address address = new Address("123 Main Street", "Apt 1", "Anytown", "CA", "99999-9999", "USA");
+
+    Customer customer =
+        new Customer(
+            UUID.randomUUID(), "Barney", "Rubble", "barney@rubble.com", "310 878 9999", address);
+
+    // Create some purchase items to add to the shopping cart
+    List<PurchaseItem> purchaseItems = new ArrayList<>();
+
+    PurchaseItem purchaseItem1 =
+        new PurchaseItem(
+            UUID.randomUUID(),
+            customer,
+            "Barry's Gourmet Peanuts Large",
+            5,
+            new BigDecimal("1"),
+            new BigDecimal("10.99"),
+            address,
+            address);
+
+    purchaseItems.add(purchaseItem1);
+
+    PurchaseItem purchaseItem2 =
+        new PurchaseItem(
+            UUID.randomUUID(),
+            customer,
+            "Barry's Gourmet Peanuts Medium",
+            3,
+            new BigDecimal("2"),
+            new BigDecimal("7.99"),
+            address,
+            address);
+
+    purchaseItems.add(purchaseItem2);
+
+    PurchaseItem purchaseItem3 =
+        new PurchaseItem(
+            UUID.randomUUID(),
+            customer,
+            "Barry's Gourmet Peanuts Small",
+            1,
+            new BigDecimal("3"),
+            new BigDecimal("4.99"),
+            address,
+            address);
+
+    purchaseItems.add(purchaseItem3);
```

Comment:
Same comments I made in the Temporal implementation:
- Move id generation into the data classes
- Avoid the purchaseItems collection. Make the addItems singular and call it once for each item.

Reply:
Please come to a consensus with @cretz about this item.

Reply:
I have my preferences, but they are not strongly held. Will discuss and get back.

Reply:
Ok, after discussion: use hardcoded IDs from the caller side instead of random for purchase items, and switch back to single-item add-item signals that you call successively (sorry for making you change this).

Reply:
addressed